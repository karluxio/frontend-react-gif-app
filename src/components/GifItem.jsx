import PropTypes from "prop-types"

export const GifItem = ({ id, title, url }) => {
  return (

    <div className="card">
      <img src={url} />
      <p>{title}</p>
    </div>

  )
}

GifItem.prototypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
