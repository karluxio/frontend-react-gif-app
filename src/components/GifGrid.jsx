import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

  const { data, loading, error } = useFetchGifs(category);

  console.log(data);

  return (
    <>
      <h3>{category}</h3>

      {loading && <h3>Loading...</h3>}

      {error && <h3>Error...</h3>}

      <div className="card-grid">
        {!loading && !error && data.map(({ id, title, images }) => (
          <GifItem
            key={id}
            id={id}
            title={title}
            url={images.downsized_medium.url}
          />
        ))}
      </div>

    </>
  )
}
