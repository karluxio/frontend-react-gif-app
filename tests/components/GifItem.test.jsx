import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('testing on GifItem component', () => {

  const title = "A great title"
  const url = "https://a-great-gif.jpg/"

  test("should match to snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot()
  })

  test("should shows title, image and url given", () => {
    render(<GifItem title={title} url={url} />)

    // screen.debug() // shows current screen rendered

    const { src, alt } = screen.getByRole("img")
    expect(src).toBe(url)
    expect(alt).toBe(alt)
  })

  test('should shows title in component', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})