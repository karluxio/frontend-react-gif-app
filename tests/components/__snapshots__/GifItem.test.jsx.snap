// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on GifItem component should match to snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://a-great-gif.jpg/"
    />
    <p>
      A great title
    </p>
  </div>
</div>
`;
